examples/asset/multi_asset_sync.rs

```diff
@@ -0,0 +1,296 @@
+        0 => asset_server.load_acquire("models/GolfBall/GolfBall.glb", guard.clone()),
+        1 => asset_server.load_acquire("models/AlienCake/alien.glb", guard.clone()),
+        2 => asset_server.load_acquire("models/AlienCake/cakeBirthday.glb", guard.clone()),
+        3 => asset_server.load_acquire("models/FlightHelmet/FlightHelmet.gltf", guard.clone()),
+        4 => asset_server.load_acquire("models/torus/torus.gltf", guard.clone()),
+        _ => unreachable!(),
+    })));
+    let future = barrier.wait_async();
+    commands.insert_resource(barrier);
+
+    let loading_state = Arc::new(AtomicBool::new(false));
+    commands.insert_resource(AsyncLoadingState(loading_state.clone()));
+
+    // await the `AssetBarrierFuture`.
+    AsyncComputeTaskPool::get()
+        .spawn(async move {
+            future.await;
+            // Notify via `AsyncLoadingState`
+            loading_state.store(true, Ordering::Release);
+        })
+        .detach();
+
+    // Display the result of async loading.
+    commands
+        .spawn(NodeBundle {
+            style: Style {
+                width: Val::Percent(100.),
+                height: Val::Percent(100.),
+                justify_content: JustifyContent::End,
+
+                ..default()
+            },
+            ..default()
+        })
+        .with_children(|b| {
+            b.spawn((
+                TextBundle {
+                    text: Text {
+                        sections: vec![TextSection {
+                            value: "Loading...".to_owned(),
+                            style: TextStyle {
+                                font_size: 64.0,
+                                color: Color::BLACK,
+                                ..Default::default()
+                            },
+                        }],
+                        justify: JustifyText::Right,
+                        ..Default::default()
+                    },
+                    ..Default::default()
+                },
+                LoadingText,
+            ));
+        });
+
+    // Camera
+    commands.spawn(Camera3dBundle {
+        transform: Transform::from_xyz(10.0, 10.0, 15.0)
+            .looking_at(Vec3::new(0.0, 0.0, 0.0), Vec3::Y),
+        ..default()
+    });
+
+    // Light
+    commands.spawn(DirectionalLightBundle {
+        transform: Transform::from_rotation(Quat::from_euler(EulerRot::ZYX, 0.0, 1.0, -PI / 4.)),
+        directional_light: DirectionalLight {
+            shadows_enabled: true,
+            ..default()
+        },
+        ..default()
+    });
+
+    // Plane
+    commands.spawn((
+        PbrBundle {
+            mesh: meshes.add(Plane3d::default().mesh().size(50000.0, 50000.0)),
+            material: materials.add(Color::srgb(0.7, 0.2, 0.2)),
+            ..default()
+        },
+        Loading,
+    ));
+}
+
+// This showcases how to wait for assets synchronously.
+fn wait_on_load(
```

Comment:
This system serves two purposes: a run-condition based on the `AssetBarrier`, and the spawning process. I think it would help clarify the `AssetBarrier` guarding stuff if you refactored this into a run condition:

```rust
fn spawn_assets(/* ... */) { /* ... */ }

fn assets_loaded(barrier: Option<Res<AssetBarrier>>) -> bool {
    // If our barrier isn't ready, return early and wait another cycle
    barrier.map(|b| b.is_ready()) == Some(true)
}

app.add_systems(spawn_assets.run_if(assets_loaded));
```

Reply:
nice suggestion!

Reply:
done!